advanced/am-pattern-2/values.yaml

```diff
@@ -0,0 +1,250 @@
+#        # add the customized configuration file under (wso2 -> deployment -> analytics -> worker -> config -> deployment.yaml)
+#        config:
+#          deployment.yaml: |-
+#            # deployment configurations for the Worker profile of WSO2 API Manager Analytics v3.0.0 (<WSO2AM_ANALYTICS>/conf/worker/deployment.yaml)
+
+      resources:
+        # These are the minimum resource recommendations for running WSO2 Stream Processor based server profiles
+        # as per official documentation (https://docs.wso2.com/display/SP440/Installation+Prerequisites).
+        requests:
+          memory: "4Gi"
+          cpu: "2000m"
+        # Please see the official documentation on WSO2 Stream Processor based Performance Analysis and Resource recommendations
+        # (https://docs.wso2.com/display/SP440/Performance+Analysis+Results) and tune the limits according to your needs
+        # where necessary.
+        limits:
+          memory: "4Gi"
+          cpu: "2000m"
+
+    am:
+      # If a custom image must be used, uncomment 'dockerRegistry' and provide its value.
+#      dockerRegistry: ""
+
+      imageName: "wso2am"
+      imageTag: "3.1.0"
+      # API Manager's Gateway specific configurations
+      gateway:
+        # Minimum number of seconds for which a newly created pod should be ready without any of its containers crashing, for it to be considered available.
+        minReadySeconds: 60
+        # Indicates whether the container is running.
+        livenessProbe:
+          # Number of seconds after the container has started before liveness probes are initiated.
+          initialDelaySeconds: 60
+          # How often (in seconds) to perform the probe.
+          periodSeconds: 10
+          # Indicates whether the container is ready to service requests.
+        readinessProbe:
+          # Number of seconds after the container has started before readiness probes are initiated.
+          initialDelaySeconds: 60
+          # How often (in seconds) to perform the probe.
+          periodSeconds: 10
+        # Number of deployment replicas
+        replicas: 2
+        strategy:
+          rollingUpdate:
+            # The maximum number of pods that can be scheduled above the desired number of pods.
+            maxSurge: 2
+            # The maximum number of pods that can be unavailable during the update.
+            maxUnavailable: 0
+#        # If the deployment configurations for the Gateway profile of WSO2 API Manager v3.0.0 (<WSO2AM>/repository/conf/deployment.toml),
+#        # add the customized configuration file under (wso2 -> deployment -> am -> gateway -> config -> deployment.toml)
+#        config:
+#          deployment.toml: |-
+#            # deployment configurations for the Gateway profile of WSO2 API Manager v3.0.0 (<WSO2AM>/repository/conf/deployment.toml)
+      # API Manager's Key Manager specific configurations
+      km:
```

Comment:
@Chathuru did you integrate support for switching between API Manager's Key Manager and IS as Key Manager, as we had done in previous versions of Kubernetes resources.

@pubudu538 do we need to integrate support for both or do we now have a recommended KM for this type of deployments?

Reply:
Better to include both.

Reply:
@pubudu538 ack.

@Chathuru please note.

Reply:
On this https://github.com/wso2/kubernetes-apim/pull/332 I only add API Manager's Key Manager. Below changes will add in a separately,
1. Add IS-as-KM
2. Add StatefulSet
3. Add Analytics HA deployment

Reply:
@Chathuru 2 can be integrated in a new release.

But improvements 1 and 3 ought to be released as part of the initial release, IMO.